third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -192,7 +200,7 @@ struct ConvertLayoutOpSwizzlingConversion
             maskSpanAffineOffset, llvmElemTy, rewriter, targetInfo);
         assert(succeeded(result));
       }
-      b.barrier();
+      targetInfo.barrier(loc, rewriter, isWarpSync);
```

Comment:
Do you need to update membar analysis to know that convert layout doesn't synchronize the whole block in these cases?

Reply:
these are the inter-rep barriers. membar analysis (correct me @Jokeren @ThomasRaoux if I'm wrong) does not assume any synchronisation coming from this operation, so it always needs to synchronise after it if any other op wants to reuse that shmem.

Reply:
we do have the assumption that if smem is used the op will have a barrier here:
https://github.com/triton-lang/triton/blob/de4376e90a3c2b5ca30ada25a50cccadeadf7f1a/lib/Analysis/Membar.cpp#L244

It sounds like we need to have a better check then?